src/main/java/me/jellysquid/mods/sodium/client/world/WorldSlice.java

```diff
@@ -321,12 +313,7 @@ public Biome getBiomeForNoiseGen(int x, int y, int z) {
      * Gets or computes the biome at the given global coordinates.
      */
     public Biome getBiome(int x, int y, int z) {
-        int relX = x - this.baseX;
-        int relY = y - this.baseY;
-        int relZ = z - this.baseZ;
-
-        return this.biomeCaches[getLocalSectionIndex(relX >> 4, relY >> 4, relZ >> 4)]
-                .getBiome(this, x, relY >> 4, z);
+        return this.world.getBiome(new BlockPos(x, y, z));
```

Comment:
Assuming `WorldSlice` instances are used on one thread at once, this should probably be replaced with a `BlockPos.Mutable` field being `.set` to the new position.

Reply:
What is better about this?